### PR TITLE
fix issues of `save_checkpoint` and test it more strictly

### PR DESCRIPTION
**Pull request by MeouSker77**
When using ipex==1.9 and custom lr_schedulers for training, if set `weights_only` to False,`save_checkpoint` method will report an error of 'Unsupport storage type' because the model is in 'xpu', so we temporarily move it to 'cpu', then move it back after `save_checkpoint`.

### CODE REVIEW

File: python/nano/notebooks/pytorch/cifar10/nano-trainer-example.ipynb [hunk 1]
```diff
@@ -2160,7 +2160,7 @@
    "source": [
     "model = LitResnet(learning_rate=0.1, batch_size=64)\n",
     "model.datamodule = data_module\n",
-    "checkpoint_callback = ModelCheckpoint(dirpath=\"checkpoints/\", save_top_k=1, monitor=\"val_loss\", filename=\"renet18_multi_ipex\", save_weights_only=True)\n",
+    "checkpoint_callback = ModelCheckpoint(dirpath=\"checkpoints/\", save_top_k=1, monitor=\"val_loss\", filename=\"renet18_multi_ipex\", save_weights_only=False)\n",
```
Comment (yangw1234) [inline python/nano/notebooks/pytorch/cifar10/nano-trainer-example.ipynb:2163]: I think we'd better extract it to a unit tests.

**Comment by yangw1234** [comment]
LGTM except one minor comment

**Comment by yangw1234** [comment]
LGTM

**Comment by MeouSker77** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/271/

**Pull request merged**